homeassistant/components/keyboard_remote/__init__.py

```diff
@@ -48,165 +60,210 @@
+
+        while True:
+            event = await self.watcher.get_event()
+            descriptor = "%s/%s" % (DEVINPUT, event.name)
+
+            descriptor_active = descriptor in self.active_handlers_by_descriptor
+
+            if (event.flags & aionotify.Flags.DELETE) and descriptor_active:
+                handler = self.active_handlers_by_descriptor[descriptor]
+                del self.active_handlers_by_descriptor[descriptor]
+                await handler.async_stop_monitoring()
+
+            elif (
+                (event.flags & aionotify.Flags.CREATE)
+                or (event.flags & aionotify.Flags.ATTRIB)
+            ) and not descriptor_active:
+                dev, handler = self.get_device_handler(descriptor)
+                if handler is None:
                     continue
+                self.active_handlers_by_descriptor[descriptor] = handler
+                await handler.async_start_monitoring(dev)
 
-            try:
-                event = self.dev.read_one()
-            except OSError:  # Keyboard Disconnected
-                self.dev = None
-                self.hass.bus.fire(
-                    KEYBOARD_REMOTE_DISCONNECTED,
-                    {
-                        DEVICE_DESCRIPTOR: self.device_descriptor,
-                        DEVICE_NAME: self.device_name,
-                    },
-                )
-                _LOGGER.debug("Keyboard disconnected, %s", self.device_id)
-                continue
+    class DeviceHandler:
+        """Manage input events using evdev with asyncio."""
 
-            if not event:
-                continue
+        def __init__(self, hass, dev_block):
+            """Fill configuration data."""
 
-            if event.type is ecodes.EV_KEY and event.value is self.key_value:
-                _LOGGER.debug(categorize(event))
-                self.hass.bus.fire(
-                    KEYBOARD_REMOTE_COMMAND_RECEIVED,
-                    {
-                        KEY_CODE: event.code,
-                        DEVICE_DESCRIPTOR: self.device_descriptor,
-                        DEVICE_NAME: self.device_name,
-                    },
-                )
+            self.hass = hass
 
+            key_types = dev_block.get(TYPE)
 
-class KeyboardRemote:
-    """Sets up one thread per device."""
+            self.key_values = set()
+            for key_type in key_types:
+                self.key_values.add(KEY_VALUE[key_type])
 
-    def __init__(self, hass, config):
-        """Construct a KeyboardRemote interface object."""
-        self.threads = []
-        for dev_block in config:
-            device_descriptor = dev_block.get(DEVICE_DESCRIPTOR)
-            device_name = dev_block.get(DEVICE_NAME)
-            key_value = KEY_VALUE.get(dev_block.get(TYPE, "key_up"))
+            self.emulate_key_hold = dev_block.get(EMULATE_KEY_HOLD)
+            self.emulate_key_hold_delay = dev_block.get(EMULATE_KEY_HOLD_DELAY)
```

Comment:
Where is this used?

Reply:
The configured delay and repeat values were supposed to be used when calling async_keyrepeat from async_monitor_input but indeed I had left the hard coded values I was using for tests by mistake, will fix.